compiler/src/dotty/tools/dotc/core/TypeApplications.scala

```diff
@@ -326,7 +325,7 @@ class TypeApplications(val self: Type) extends AnyVal {
       case dealiased: HKTypeLambda =>
         def tryReduce =
           if (!args.exists(isBounds)) {
-            val followAlias = Config.simplifyApplications && {
+            val followAlias = (Config.simplifyApplications || self.typeSymbol.isPrivate) && {
```

Comment:
We still need to simplify type applications if they are type aliases that are declared `private` since we lose access to the type alias' symbol at the later point, rendering the new approach in `isSubType` inapplicable. I discovered this through a test case in the "endpoints4s" community build project that uses Akka Http, which has this definition: https://github.com/akka/akka-http/blob/v10.2.8/akka-http/src/main/scala/akka/http/scaladsl/marshalling/PredefinedToResponseMarshallers.scala#L23. I wonder whether this code should be legal since the private type escapes its scope?

Reply:
>  I wonder whether this code should be legal since the private type escapes its scope?

Depends on where the private type alias appears: https://github.com/lampepfl/dotty/blob/e05af52bcf9f5cfcbda532c6dfaee4deac9928ac/compiler/src/dotty/tools/dotc/typer/Checking.scala#L565-L585

Reply:
Ah, I see. I wasn't aware of that but this makes sense.